bolt-modules/file/lib/puppet/functions/file/write.rb

```diff
@@ -15,8 +15,15 @@
   end
 
   def write(filename, content)
+    executor = Puppet.lookup(:bolt_executor)
```

Comment:
```suggestion
    executor = Puppet.lookup(:bolt_executor) {}
```
This will error if we don't provide it a default.